### DISCUSSION TITLE
Conventional commit

### DISCUSSION THREAD

**Discussion by zippeurfou** [NeogitOrg/neogit#1246] [Ideas]
It would be great if we could set up option for commit template. For example using conventional commit.
It does not need to be overly complicated. For example we could start with adapting https://github.com/olacin/telescope-cc.nvim or having an integration with https://github.com/commitizen/cz-cli.

**Comment by CKolkey** [comment]
Does the telescope extension not work? It looks like it can use the git CLI, which we will see update the index and can refresh. Make sure the file watcher is enabled, or use `nightly` branch where its enabled by default

**Reply by zippeurfou** [reply]
Yes it does work like a charm. It defeats a bit the purpose a bit of how I was intended to use neogit where neogit allows you to do much more than just setup a git commit message and I want to use it as my main tool for git. This is why I was suggesting as a feature request so I don't have to stage with one tool and commit with another. In addition, not all repositories use conventional commit so now I would find myself in a scenario where sometimes I use the telescope extension, sometimes I don't and use neogit commit which I think is a bit cumbersome. 
Finally, the idea can extends beyond conventional commit where one could imagine creating git commit template per project or globally and "magically" hooking neogit to it so you can use best practice in your git commit messages directly in neogit.

**Reply by CKolkey** [reply]
Not sure the vision you have, but I'd gladly look at a PR if you are up for it :)

**Reply by Dich0tomy** [reply]
Was there any though about alternative highlighting for conventional commits tho:
`type[(scope)][!]: message`
type, scope, ! and message could be highlighted *slightly* different
